### PR TITLE
Deallocate xmlNode from xmlCopyNode in QMCMain::m_qmcaction

### PR DESCRIPTION
**Pull request by williamfgc**
## Proposed changes

Deallocate `xmlNodePtr` in `QMCMain::m_qmcation`  container only if the memory is owned before clearing.
Fixes a leak that propagates through several deterministic tests as reported in #3312

## What type(s) of changes does this code introduce?

- Bugfix
- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
Ubuntu 20.04, must pass CI

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/QMCApp/QMCMain.cpp [hunk 1]
```diff
@@ -260,6 +260,14 @@ bool QMCMain::execute()
       break;
     }
   }
+  // free if m_qmcation owns the memory of xmlNodePtr before clearing
+  for (auto& qmcactionPair : m_qmcaction)
+  {
+    if (!qmcactionPair.second)
+    {
```
Comment (ye-luo) [inline src/QMCApp/QMCMain.cpp:267]: Can we remove both pair of brackets?

**Comment by ye-luo** [comment]
Test this please

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

**Review by prckent — APPROVED** [review]
Thanks William.

**Pull request merged**